### ISSUE TITLE
Type 'Rule[]' has no properties in common with type - Wrong validator example

### ISSUE THREAD

**Issue by backspacerhino** [adonisjs/adonis-validation-provider#73]
## Package version
5.0.0-preview

## Node.js and npm version
v12.14.1 and 6.13.4

## Sample Code (to reproduce the issue)
`node ace make:validator TestValidator` will create validator with an example:

>    schema.string([ rules.alpha() ])

but TS complains. From [test](https://github.com/adonisjs/validator/blob/develop/test/schema.spec.ts) for Schema I can see that it should instead be

>    schema.string({},[ rules.alpha() ])

Once I do it like that I no longer get TS complains and everything works.

**Comment by agtabesh** [comment]
According to the code, because of existence of `rules` parameter in all `schema type`s, I think it's better to have `rules` as first argument.

**Comment by thetutlage** [comment]
You may not always want to define rules on a schema type. For example:

```ts
{
  name: schema.string()
}
```

or 

```ts
{
  name: schema.string({ trim: true })
}
```

So having the rules array at first position is very subjective. I am okay keeping it as a last argument.

**Comment by thetutlage** [comment]
Back to the real issue. It's been fixed in `@adonisjs/assembler@1.3.4`. Feel free to update the dependency and should be good
